modules/ai/site-planner-connect/module.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Elementor\Modules\Ai\SitePlannerConnect;
+
+defined( 'ABSPATH' ) || exit;
+
+class Module {
+
+	public function __construct() {
+		add_action( 'rest_api_init', [ $this, 'on_rest_init' ] );
+		add_action( 'admin_menu', [ $this, 'register_menu_page' ], 100 );
+		add_filter('rest_prepare_application_password', function ( $response, $item, $request ) {
+			if ( $request->get_route() === '/wp/v2/users/me/application-passwords' && is_user_logged_in() ) {
+				$user = wp_get_current_user();
+				$response->data['user_login'] = $user->user_login;
+			}
+			return $response;
+		}, 10, 3);
+	}
+
+	public function on_rest_init(): void {
+		( new WpRestApi() )->register();
+	}
+
+	public function register_menu_page() {
+		add_submenu_page(
+			null, // Hidden page
+			'App Password Generator',
+			'App Password',
+			'manage_options',
+			'site-planner-password-generator',
+			[ $this, 'render_menu_page' ]
+		);
+	}
+
+	public function render_menu_page() {
+		$root_url = 'https://planner.elementor.com';
```

Comment:
Shouldn't this value come from a parameter? should this DNS name change, It would be difficult to change, and yes, I understand that it's not supposed to change, but as a best practice. wdut? Other then that it's a different code smell called magic string/number

Reply:
Do you mean a remote config param?
The user may still not be connected to a remote service in this flow.

Reply:
No, I meant the actual value - is it a configuration? The DNS name might change, no? and then u will have to search the whole project for these instances, rather then a single page - easy to maintain